core/app/waste_item/application/use_cases/get_items_by_userId.py

```diff
@@ -0,0 +1,13 @@
+from core.app.waste_item.domain.ports import WasteItemRepository
+from core.app.waste_item.application.exceptions import WasteItemNotFoundException
+
+class GetItemsByUserIdUseCase:
+    def __init__(self, waste_item_repository: WasteItemRepository):
+        self.waste_item_repository = waste_item_repository
+
+    def execute(self, user_id: str):
+        items = self.waste_item_repository.list_by_user_id(user_id)
+        if not items:
+            raise WasteItemNotFoundException()
+        
+        return items
```

Comment:
First of all, you should not raise an exception if the items are empty, you should just return an empty list.

Second of all, you should instead raise an exception if the user_id is not found

Third of all, You should not only raise the exception, but write a descriptive error message

And last but not less important, you should always write the response type definition, is a good practice we have not followed so strictly but is a good practice
```suggestion
from typing import List
from app.domain.entities import WasteItem

class GetItemsByUserIdUseCase:
    def __init__(self, waste_item_repository: WasteItemRepository):
        self.waste_item_repository = waste_item_repository

    def execute(self, user_id: str) -> List[WasteItem]:
        if not self.waste_item_repository.user_exists(user_id):
            raise UserNotFoundException(f"User with ID '{user_id}' not found.")

        items = self.waste_item_repository.list_by_user_id(user_id)
        return items  # Empty list is fine
```